lib/resources/nginx_module.rb

```diff
@@ -0,0 +1,35 @@
+# encoding: utf-8
+# author: Rony Xavier, rx294@gmail.com
+# author: Aaron Lippold, lippold@gmail.com
+
+module Inspec::Resources
+  class NginxModule < Inspec.resource(1)
+    name 'nginx_module'
+    desc 'Use the nginx_module InSpec audit resource to test nginx modules on Linux platforms.'
+    example "
+      describe nginx_module('http_auth_request') do
+        it { should be_loaded }
+      end
+      "
+
+    def initialize(modulename = nil, nginx_path = nil)
+      @module = modulename
+      @nginx_path = nginx_path
+      return skip_resource 'The `nginx_module` resource is not yet available on your OS.' if !inspec.os.linux?
+    end
+
+    def loaded?
```

Comment:
Since a user does not have to supply a module, then this method could get funky... what if the user does this?

```
describe nginx_module do
  it { should be_loaded }
end
```

It will try to match on nil and fail, but it won't be a great experience.

Even thought it will cause InSpec to completely unwind and not run the remaining tests, this feels like a syntax issue that should get raised. I'd consider raising an exception here:

```
def loaded?
  raise "No module name supplied - cannot check if nginx module is loaded" if @module_name.nil?
  # rest of method...
end
```